### PR TITLE
BE: Profile Model

### PR DESCRIPTION
**Pull request by dproc96**
**What it does**
- Create a DB model for the Profile
- Create test cases for the Profile model
- Create file for future test cases for the Database
- Create index.js for grouping models (duplicated work in PR #24)

**What to look for**
- Any validators I should add to the Model
- Any statics or methods that would be useful to add
- Any test cases I should add

**What to avoid**
- I used my instincts for what should be required, if more data should be required I'll add validators for that too

### CODE REVIEW

**Comment by Braagaa** [comment]
Maybe we can add an age restriction validator to the birthDay attribute. Or to check and make sure that the availability end date is not less than the beginning availability date. Anyone else agree?

**Comment by dproc96** [comment]
> Maybe we can add an age restriction validator to the birthDay attribute. Or to check and make sure that the availability end date is not less than the beginning availability date. Anyone else agree?

Oh those are good, I'll add those in! Thanks Michael!

**Comment by dproc96** [comment]
@Braagaa I've added your suggestions to the code and they're working!

**Comment by Braagaa** [comment]
:+1:

**Comment by Braagaa** [comment]
I think we missing some attributes. The mocks are showing email, address, and phone number. We possibly need a profile image url as well.

**Comment by dahrk** [comment]
Michael B made a good point, there are some more fields that can be added.
A style tip would be to add newlines at the end of your files

File: server/test/Database.spec.js [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            throw err
+          }
+          if (!profile) {
+            throw new Error('No data found')
+          }
+          done()
+        })
+      })
+
+      it("Rejects user if under 18", (done) => {
```
Comment (dahrk) [inline server/test/Database.spec.js:189]: Something to verify - if no date is entered, will the user be denied? If this check is being run, should birthdate be required as well?

**Review by dahrk — APPROVED** [review]

**Review by aeciorc — COMMENTED** [review]
Looks good Dan. One thing we need to discuss better is the distinction between a regular user and a sitter. For a regular user, for example, the availability wouldn't be required. We can talk about that during our standup

**Comment by dproc96** [comment]
@Braagaa @dahrk Yeah, those would be good things to include, I was following the ticket as written but we can discuss modifications

**Review by aeciorc — APPROVED** [review]

**Pull request merged**